### DISCUSSION TITLE
Ошибка при обработке недоступных треков из Cache

### DISCUSSION THREAD

**Discussion by Bondian** [Chimildic/goofy#130] [Помощь с ошибкой] [answered]
Давно работающая функция вдруг стала останавливаться на ошибке при удалении недоступных треков. Cache-файл предварительно сжат. 
Вот в этом месте выдаёт ошибку:
```
  let rusTracks = Cache.read('rus.json');
  Filter.removeUnavailable(rusTracks);
```
14:08:10	Примечание	Выполнение начато
14:08:25	Информация	Пауза в отправке запросов 11
14:08:38	Информация	Пауза в отправке запросов 17
14:08:57	Информация	Трек нельзя послушать: 7FaNzcQwtXUjiZlKXJUTyQ - slider magnit baby tonight
14:08:46	Ошибка	
TypeError: Cannot read property 'hasOwnProperty' of null
(функция без названия)	@ library.gs:1092
defineUnavailableState	@ library.gs:1091
removeUnavailable	@ library.gs:1076
myExhbhgvfgFunction	@ Без названия.gs:4

Попробовал подставить посторонний Cache-файл с недоступными треками - ошибка не появляется.
Выполнил перезапуск функции, которая записывает Cache-файл, на котором появляется ошибка, предварительно его сжав:
```
  Cache.compressTracks(rusTracks);
  Cache.write('rus.json', rusTracks);
```
С обновлённым Cache-файлом появляется та же ошибка...

**Comment by Chimildic** [comment]
Прикрепите файл кэша. Он находится на [гугл диске](https://drive.google.com/drive/my-drive) в папке `goofy data`.

**Comment by Bondian** [comment]
[rus.zip](https://github.com/Chimildic/goofy/files/7267408/rus.zip)

**Comment by Chimildic** [comment, marked as answer]
Поскольку кэш лежит долго, данные в нем устаревают. Трека с `id = 2kuSspCmyRw3iM0nsYWrq4 (niletto все решено)` больше не существует. Такой случай не был предусмотрен, поэтому падает в ошибку. 

Как относительно недавно выяснилось, Spotify может менять _id_, но не частно. Поэтому постепенно ловим такие ошибки.
Попробуйте обновить библиотеку сейчас, должно помочь. В логах будет соответствующее сообщение.

**Reply by Bondian** [reply]
Большое спасибо. Как всегда - всё по полочкам... Как только доберусь до ПК (возможно  только через день) - сразу обновлю библиотеку...
